### DISCUSSION TITLE
Download feature - Password Protected

### DISCUSSION THREAD

**Discussion by SSj-Saturn** [LycheeOrg/Lychee#2529] [Ideas]
### Enhancement

I know there's a feature that allows you password protect an album if they're not logged in, but I'd like anonymous people to view the photos, but require a password to download them. Take a lot of photos for events and this would solve a lot of issues.

**Comment by ildyria** [comment]
I need to see how to implement this.
There might be more interesting implementations like setting a system of basket etc.

**Comment by SSj-Saturn** [comment]
Neat, definitely interested in either way to set a password to download feature for a whole folder and contents, or per specific folder.

I honestly have concerns about AI scraping, or just creeps, so this provides a little bit of security for getting the full quality version.
Just hard to incentivize people to create accounts, and then I manually have to give access to each folder etc. but even then they get full download access to any folders etc etc.

So setting an event password makes it much easier (and hopefully the analytics tracking via GA gives enough information to see usage etc).

Appreciate Lychee and everything you do! The best solution for what I need!

**Reply by ildyria** [reply]
Just to clarify one thing.

> I honestly have concerns about AI scraping, or just creeps, so this provides a little bit of security for getting the full quality version.

As long as you do not tick `Downloadable` and `Original` neither are available.
Also, you cannot prevent tech-savy people from downloading the images. Right click, even if disabled is by-passable by loading the developers tools in the web browser.

Am I understanding right that you would like those (`originals/downloadables`) still to be available but only with an additional protection.

**Reply by SSj-Saturn** [reply]
Agreed, if anonymous users want to grab images, they'll find a way via scraping and devtools etc, but it'll be the small web version and not the original size.

I'd love to have the Download option apply to the Original sized version for full quality if possible.

Hopefully that makes sense.
